plugins/follow_me/follow_me.h

```diff
@@ -0,0 +1,170 @@
+ * accordingly; this results in drone movement relative to that of GCS.
+ *
+ * Please refer https://docs.px4.io/en/flight_modes/follow_me.html for more information.
+ *
+ */
+class FollowMe
+{
+public:
+    /**
+     * @brief Constructor (internal use only).
+     * @param impl Private internal implementation.
+     */
+    explicit FollowMe(FollowMeImpl *impl);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~FollowMe();
+
+    /**
+     * @brief Results of FollowMe operations
+     */
+    enum class Result {
+        SUCCESS = 0, /**< @brief %Request succeeded. */
+        NO_DEVICE, /**< @brief No device connected. */
+        CONNECTION_ERROR, /**< @brief %Connection error. */
+        BUSY, /**< @brief Vehicle busy. */
+        COMMAND_DENIED, /**< @brief Command denied. */
+        TIMEOUT, /**< @brief %Request timeout. */
+        NOT_ACTIVE, /**< @brief FollowMe is not activated yet */
+        UNKNOWN /**< @brief Unknown error. */
+    };
+
+    /**
+     * @brief FollowMe Configuration.
+     * Reference: https://docs.px4.io/en/advanced_config/parameter_reference.html#follow-target
+     */
+    struct Config {
+        /**
+         * @brief Side to follow target from.
+         */
+        enum class FollowDirection {
+            FRONT_RIGHT, /**< @brief Follow from front right. */
+            BEHIND, /**< @brief Follow from behind. */
+            FRONT, /**< @brief Follow from front. */
+            FRONT_LEFT, /**< @brief Follow from front left. */
+            NONE
+        };
+        // Default Configuration
+        static const float DEF_HEIGHT_M; /**< @brief Default height above home: 8.0 mts */
+        static const float DEF_FOLLOW_DIST_M; /**< @brief Default Follow distance: 8.0 mts */
+        static const FollowDirection DEF_FOLLOW_DIR; /**< @brief Default follow direction: BEHIND */
+        static const float DEF_RESPONSIVENSS; /**< @brief Default responsiveness to target: 0.5 */
+
+        // Min & Max ranges as per PX4 FollowMe Configuration.
+        static const float MIN_HEIGHT_M; /**< @brief Min follow height, in mts. >= 8.0m */
+        static const float MIN_FOLLOW_DIST_M;  /**< @brief Min follow distance from, in mts. >= 1.0m */
+        static const float MIN_RESPONSIVENESS; /**< @brief Min responsiveness: 0.0 */
+        static const float MAX_RESPONSIVENESS; /**< @brief Max responsiveness: 1.0 */
+
+        float min_height_m = DEF_HEIGHT_M; /**< @brief Min follow height above home */
+        float follow_dist_m =
+            DEF_FOLLOW_DIST_M; /**< @brief Follow horizontal distance b/w Vehicle & GCS. */
+        FollowDirection follow_dir = DEF_FOLLOW_DIR; /**< @brief Side to follow target from. */
+        float responsiveness = DEF_RESPONSIVENSS; /**< @brief Responsiveness: Range (0.0 to 1.0)*/
+
+        /**
+         * @brief Human-readable string of enum `FollowDirection`.
+         * @param dir follow direction
```

Comment:
Capital F.
Are you allowed to use shortened names like `dir` for parameters?

Reply:
I was like "yes dir for directory is fair" and then realized it was direction not directory. I vote to spell it out.